### PR TITLE
Multi entity edition

### PR DESCRIPTION
**Pull request by matthewdylan**
#### What's this PR do?
Adds functionality to make the app more dynamic and usable by different news orgs

#### Why are we doing this? How does it help us?
This is a first step in creating a one-for-all donation app for our different news entities.

#### How should this be manually tested?
For TT:
Visit the [staging site](https://donations-testing.herokuapp.com/donate)
Ensure the text looks as expected (nothing about Waco mentioned)
Attempt giving a donation (any amount/interval works)
Ensure our TT stripe account (in test mode) lists the new donation under "Transactions"
Ensure the donation was rippled to salesforce and has "texas" set as the newsroom

For WB:
Same as above except staging site is [here](https://waco-donations-2cb57b3c7271.herokuapp.com/donate), obviously the text should reflect Waco, look at the Waco Bridge account in stripe and the newsroom field in SF should be marked "waco"

#### How should this change be communicated to end users?
I'll keep the #local-waco channel up to date on progress

#### Are there any smells or added technical debt to note?
The /waco page needs to be removed and a handful of things can follow. There is more work to do to make this app fully dynamic but it would involve abstracting out some of the text bits in ways we haven't decided on yet.

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recsChD2Q2ybfU8eT?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [ ] Added automated tests? *( )*
* [ ] Tested manually on mobile? *( )*
* [ ] Checked BrowserStack? *( )*
* [ ] Checked for performance implications? *( )*
* [ ] Checked accessibility? *( )*
* [ ] Checked for security implications? *( )*
* [ ] Updated the documentation/wiki? *( )*

#### TODOs / next steps:

* [ ] *your TODO here*

### CODE REVIEW

**Review by ashley-hebler — COMMENTED** [review]
At a high level, this is working great!

In my mind, it feels cleanest to release it in this order:
1. Multi site - introduce feature to send newsroom info to stripe/sf
    - this is essentially this PR after putting /waco and the campaign id rule back
     - have that running live for support.tt.org
     - confirm everything looks good on Anna's end after a few donations come through
3. Waco site - introduce new template updates (header/footer etc.) for waco
     - deploy that to both tt and waco
(deploy waco to new domain on the 25th) 
4. Drop /waco path
     - now that we know we don't need that page to collect donations anymore, we can safely remove that route
5. Meet to discuss /circle and /business for waco

I know we chatted about that but wanted to clearly state that here. 


Question: do we need to bulk update all existing RDOs and opportunities with newsroom=texas?

**Comment by matthewdylan** [comment]
> At a high level, this is working great!
> 
> In my mind, it feels cleanest to release it in this order:
> 
> 1. Multi site - introduce feature to send newsroom info to stripe/sf
>    
>    * this is essentially this PR after putting /waco and the campaign id rule back
>    * have that running live for support.tt.org
>    * confirm everything looks good on Anna's end after a few donations come through
> 2. Waco site - introduce new template updates (header/footer etc.) for waco
>    
>    * deploy that to both tt and waco
>      (deploy waco to new domain on the 25th)
> 3. Drop /waco path
>    
>    * now that we know we don't need that page to collect donations anymore, we can safely remove that route
> 4. Meet to discuss /circle and /business for waco
> 
> I know we chatted about that but wanted to clearly state that here.
> 
> Question: do we need to bulk update all existing RDOs and opportunities with newsroom=texas?



> At a high level, this is working great!
> 
> In my mind, it feels cleanest to release it in this order:
> 
> 1. Multi site - introduce feature to send newsroom info to stripe/sf
>    
>    * this is essentially this PR after putting /waco and the campaign id rule back
>    * have that running live for support.tt.org
>    * confirm everything looks good on Anna's end after a few donations come through
> 2. Waco site - introduce new template updates (header/footer etc.) for waco
>    
>    * deploy that to both tt and waco
>      (deploy waco to new domain on the 25th)
> 3. Drop /waco path
>    
>    * now that we know we don't need that page to collect donations anymore, we can safely remove that route
> 4. Meet to discuss /circle and /business for waco
> 
> I know we chatted about that but wanted to clearly state that here.
> 
> Question: do we need to bulk update all existing RDOs and opportunities with newsroom=texas?

Yep, sounds great. One other thing to note, I moved the linting fixes over to [this pr](https://github.com/texastribune/donations/pull/1211) so those aren't held up and the app can be right with the world again.

**Review by ashley-hebler — APPROVED** [review]
Hey @matthewdylan, thanks for pulling out the linting fixes. Good call.

My last round of testing all went well I think so we're good to deploy.

One nitpick/polish question: 
Screenshot 2024-09-16 at 1 29 51 PM

Screenshot 2024-09-16 at 1 30 46 PM

I noticed that for the Newsroom value in Salesforce is either "texas" or "Waco". I wonder if that capitalization discrepancy will confuse us down the road in terms of all the string conditions we are depending on e.g. `if NEWSROOM["name"] == "waco"`.


There's one open question regarding previous data from my last review note that I'll just ask you directly about on Slack

**Comment by matthewdylan** [comment]
> Hey @matthewdylan, thanks for pulling out the linting fixes. Good call.
> 
> My last round of testing all went well I think so we're good to deploy.
> 
> One nitpick/polish question: Screenshot 2024-09-16 at 1 29 51 PM
> 
> Screenshot 2024-09-16 at 1 30 46 PM
> 
> I noticed that for the Newsroom value in Salesforce is either "texas" or "Waco". I wonder if that capitalization discrepancy will confuse us down the road in terms of all the string conditions we are depending on e.g. `if NEWSROOM["name"] == "waco"`.
> 
> There's one open question regarding previous data from my last review note that I'll just ask you directly about on Slack

Very odd, you can tell in stripe that it's coming through lowercase so I'm not sure what's going on... I adder .lower() at the point where the val is grabbed from the stripe event and pushed to sf.

File: server/app.py [hunk 1]
```diff
@@ -549,6 +544,9 @@ def validate_form(FormType, bundles, template, function=add_donation.delay):
 
     if FormType is DonateForm:
         donation_type = "membership"
+        # TODO discuss when we don't need to default to the WACO_CAMPAIGN_ID
```
[the remark this answers is not in the document]
Comment (matthewdylan) [reply server/app.py:554]: Yeah, makes sense to me.

**Comment by matthewdylan** [comment]
> > Hey @matthewdylan, thanks for pulling out the linting fixes. Good call.
> > My last round of testing all went well I think so we're good to deploy.
> > One nitpick/polish question: Screenshot 2024-09-16 at 1 29 51 PM
> > Screenshot 2024-09-16 at 1 30 46 PM
> > I noticed that for the Newsroom value in Salesforce is either "texas" or "Waco". I wonder if that capitalization discrepancy will confuse us down the road in terms of all the string conditions we are depending on e.g. `if NEWSROOM["name"] == "waco"`.
> > There's one open question regarding previous data from my last review note that I'll just ask you directly about on Slack
> 
> Very odd, you can tell in stripe that it's coming through lowercase so I'm not sure what's going on... I adder .lower() at the point where the val is grabbed from the stripe event and pushed to sf.

Follow up... that .lower() didn't seem to do anything. Think this is on the SF side so we reached out to Anna.

**Pull request merged**